processor/src/processor/operations/crypto_ops.rs

```diff
@@ -200,6 +200,63 @@ pub(super) fn op_horner_eval_base<P: Processor>(
     Ok(P::HelperRegisters::op_horner_eval_registers(alpha, k0, k1, acc_tmp))
 }
 
+// LOG PRECOMPILE OPERATION
+// ================================================================================================
+
+/// Logs a precompile event by absorbing `TAG` and `COMM` into the precompile sponge
+/// capacity.
+///
+/// Stack transition:
+/// `[COMM, TAG, PAD, ...] -> [R1, R0, CAP_NEXT, ...]`
+///
+/// Where:
+/// - The hasher computes: `[CAP_NEXT, R0, R1] = Rpo([CAP_PREV, TAG, COMM])`
+/// - `CAP_PREV` is the previous sponge capacity provided non-deterministically via helper
+///   registers.
+#[inline(always)]
+pub(super) fn op_log_precompile<P: Processor>(
+    processor: &mut P,
+    tracer: &mut impl Tracer,
+) -> [Felt; NUM_USER_OP_HELPERS] {
+    // Read TAG and COMM from stack
+    let comm = processor.stack().get_word(0);
+    let tag = processor.stack().get_word(4);
+
+    // Get the current precompile sponge capacity
+    let cap_prev = processor.precompile_transcript_state();
+
+    // Build the full 12-element hasher state for RPO permutation
```

Comment:
The comment mentions writing [CAP_NEXT, R0, R1] but the actual order based on the iteration is [R1, R0, CAP_NEXT] due to the reverse indexing. The comment should accurately reflect the actual stack layout.